java/ext/javadoctest.py

```diff
@@ -0,0 +1,57 @@
+import subprocess
+import os
+from sphinx.ext.doctest import TestcodeDirective, TestoutputDirective, DocTestBuilder, Any, Dict, doctest, sphinx
+from sphinx.locale import __
+from subprocess import Popen, PIPE
+
+class JavaDocTestBuilder(DocTestBuilder):
+    """
+    Runs java test snippets in the documentation.
+    """
+    name = 'javadoctest'
+    epilog = __('Java testing of doctests in the sources finished, look at the '
+                'results in %(outdir)s/output.txt.')
+    def compile(self, code: str, name: str, type: str, flags: Any, dont_inherit: bool) -> Any:
+        # go to project that contains all your arrow maven dependencies
+        dir_arrow_maven_project = os.path.dirname(os.getcwd())+'/java/source/demo'
```

Comment:
We should try to use pathlib or at least os.path.join throughout.

Reply:
Changed